src/sim/insttracer.hh

```diff
@@ -232,14 +232,38 @@ class InstRecord
     setData(const RegClass &reg_class, RegVal val)
     {
         new(&data.asReg) InstResult(reg_class, val);
-        dataStatus = DataReg;
+        switch (reg_class.type()) {
+            case IntRegClass:
+            case MiscRegClass:
```

Comment:
I'm not really confident about this part. I'm not entirely sure that MiscRegs and CCRegs are always not larger than 64 bits.